### ISSUE TITLE
Refactor Select in Accordance with W3C Authoring Practices Guidelines

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#4765]
closes #4452 
closes #4373 

Our current implementation of Select has serious screen reader accessibility issues. A refactor getting it back in line with expected behavior and functionality of default browser `<select>` elements as seen [here in an MDN example](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select) or [the Authoring Practices guidelines for a collapsible listbox](https://www.w3.org/TR/2019/NOTE-wai-aria-practices-1.1-20190814/examples/listbox/listbox-collapsible.html) is our path to compliance. 

This would involve a re-evaluation of certain non-standard _advanced functionality_ like categories -- which currently break screen reader accessibility in VoiceOver and others.

We should also take this opportunity to very clearly state the differences and intended usage for Select vs Dropdown vs Combobox since some of their shared features make this hard to intuit (especially from the dev side on the React documentation site).

**Comment by joshblack** [comment]
This is blocked as we need design guidance to figure out how to address category support. It seems like the native select does not support this, but we can handle it [inside of a combobox](https://dequelabs.github.io/combobo/demo/).
